### PR TITLE
Add new mapping function and fix some broken tests

### PR DESCRIPTION
**Pull request by spinwang**
Add 
* valueUnit, to get value unit from the string
* getKey, to get the data field key

Fix some broken tests. You can found the broken tests by running `yarn test`

### CODE REVIEW

File: functions.js [hunk 1]
```diff
@@ -87,6 +123,33 @@ function string(value) {
...
+
+  value = number(value);
+
+  return {
+    value,
+    unit
+  }
+}
+
+function getKey(value, key) {
```
Comment (punya) [inline functions.js:149]: Could you add a comment describing why someone would want to use this function?

Comment (spinwang) [reply functions.js:159]: added comments, thanks for the suggestion!

**Comment by punya** [comment]
Please update this PR on top of latest master. All tests now pass and Travis is enabled.

File: functions.js [hunk 2]
```diff
@@ -64,13 +65,43 @@ const min =
...
+ * @param input
+ * @return {{metric: string, unit: string}}
+ */
+function extractMetricAndUnitViaBracket(input) {
+  if (!_.isString(input)) {
+    throw new Error(`${input} is not a string`);
+  }
+  const regexp = /^(.+)(?:\s*)(?:\((.+)\))$/;
+  const components = input.match(regexp);
+  if (_.isNull(components)) {
```
Comment (chunfengd) [inline functions.js:91]: can you use `_.isNil`? or simply `!compontents`

File: functions.js [hunk 3]
```diff
@@ -64,13 +65,43 @@ const min =
...
+    return {
+      metric: input,
+      unit: null,
+    };
+  }
+  return {
+    metric: trim(components[1]),
+    unit: _.isUndefined(components[2]) ? null : components[2],
```
Comment (chunfengd) [inline functions.js:99]: I would use `components[2] || null`

**Pull request closed without merging**